xc/xenops_server_xen.ml

```diff
@@ -2855,6 +2871,20 @@ module Actions = struct
 						None in
 				Opt.iter (fun x -> Updates.add x internal_updates) update in
 
+		let fire_event_on_qemu domid signal =
+			let d = int_of_string domid in
+			let open Xenstore_watch in
+			if not(IntMap.mem d domains)
+			then debug "Ignoring qemu-pid-signal watch on shutdown domain %d" d
+			else
+				debug "Received unexpected qemu-id-signal %s for domid %d" signal d;
+				let di = IntMap.find d domains in
+				let open Xenctrl in
```

Comment:
Why do you need to open this? I don't think that you are using anything from this module?